crates/swift-integration-tests/src/result.rs

```diff
@@ -41,6 +37,15 @@ mod ffi {
         fn val(&self) -> u32;
     }
 
+    #[swift_bridge(swift_repr = "struct")]
+    struct ResultTestTransparentStruct(pub String);
+
+    extern "Rust" {
+        fn rust_func_returns_result_null_transparent_struct(
+            succeed: bool,
+        ) -> Result<(), ResultTestTransparentStruct>;
+    }
```

Comment:
As it stands now it's unclear why this is here. A maintainer 5 years from now would look at this and be confused as to why we aren't calling it from the Swift side.
This is here for a special reason (to prevent regressions when we remove this `#[allow(unused)]`, but this will not be clear to a maintainer in 5 years)

---

Let's explain why this was put here.

- In Rust `1.79.0` we noticed a warning when we returned `-> Result<*, TransparentStruct>` when the struct has at least one field
- This is a regression in Rust that should be fixed in a future version
- At some point after that regression is fixed we'll remove the `#[allow(unused)]` from the coegen
- This test is meant to help us be sure that after removing the `#[allow(unused)]` the warning does not come back
- We do not call this function from `Swift` since only exists here as a test that no warning is generated in Rust

Reply:
Funny thing... I didn't even see the Swift integration tests. I'll do one better and add a test for results with transparent structs there. It seems like I'm probably not the only person who will want to use that feature.

Reply:
That works, but if that's the case need to also add a codegen test for a Rust function that returns `-> Result<TransparentStruct, TransparentStruct>`

Similar to this but for transparent structs https://github.com/chinedufn/swift-bridge/blob/852ece728d3f362f3ada4368e037fe9ebb2a913f/crates/swift-bridge-ir/src/codegen/codegen_tests/result_codegen_tests.rs#L129-L200

Anytime we add support for a new type we add a codegen test and an integration test.

In this case the new type that we are explicitly supporting (now that you're adding this test) is `-> Result<TransStruct, TransStruct>`